pilot/pkg/bootstrap/sidecarinjector.go

```diff
@@ -90,20 +89,17 @@ func (s *Server) initSidecarInjector(args *PilotArgs) (*inject.Webhook, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to create injection webhook: %v", err)
 	}
-	// Patch cert if a webhook config name is provided.
-	// This requires RBAC permissions - a low-priv Istiod should not attempt to patch but rely on
+	// Patching certs requires RBAC permissions, low-priv Istiod should not attempt to patch but rely on
 	// operator or CI/CD
-	if features.InjectionWebhookConfigName.Get() != "" {
```

Comment:
Sorry I am contradicting myself, but  I remembered currently in order to stop the patching, we are setting INJECTION_WEBHOOK_CONFIG_NAME="". This is for cases where the webhook is behind a trusted cert so we do not need to patch it. So maybe we still need this, but just for this check and not in the actual code. It really should be more like "enable webhook" but we can keep the name for backwards compat?

Reply:
And we don't need it in the templates, so your changes there still make sense

Reply:
Added the code references to `features.InjectionWebhookConfigName` back in